### ISSUE TITLE
SST 39VF040 - IC is not recognized

### ISSUE THREAD

**Issue by p-b-p** [jdfr228/PS1-Disc-Based-Game-ID#4]
It is flashed with stock SCPH-9001 BIOS and everything is working:

![20230607_091057](https://github.com/jdfr228/PS1-Disc-Based-Game-ID/assets/95297132/df8d931a-2e63-48b5-9eb0-119fc07ba0de)

When /WE pin is connected to VCC if I boot the flasher CD and press L1 the IC is detected as:

![20230607_002855](https://github.com/jdfr228/PS1-Disc-Based-Game-ID/assets/95297132/99f91c82-f257-4196-9cd7-f960bdcae9e2)

When /WE is connected to /SWR0 or /WR:

![image](https://github.com/jdfr228/PS1-Disc-Based-Game-ID/assets/95297132/e0067d28-a65f-44f3-af77-ed2ac25f0cec)

the flasher CD boots, but I am unable to interact and detect the IC by pressing L1. It is not working and nothing happens:

![20230607_002329](https://github.com/jdfr228/PS1-Disc-Based-Game-ID/assets/95297132/6c42f866-7a2e-4c63-aeb0-b8b58e11716b)

IC pins are checked for shortages and stock BIOS is working as expected - memory card menu is working and I am able to boot fine backup CDs. What could be wrong with my setup?

IC's datasheet: https://datasheet.lcsc.com/lcsc/2207211800_Microchip-Tech-SST39VF040-70-4I-WHE_C633807.pdf

**Comment by jdfr228** [comment]
Hi p-b-p,

This is matching what I've heard from another user testing the flashing software- IC 13 00 when /WE is connected to VCC, Not Detected when /WE is connected to /SWR0 (CPU pin 100).  So I doubt you have a short or anything.  I also have the Not Detected message which is where I had to stop testing on my personal system for the time being.

BF D7 is the expected ID for the SST-39VF040, which is what would indicate everything is working properly.

In talking to megavolt85, who wrote the software, pin 22 on the BIOS (/CE) may need to also be isolated from the mainboard (easy with your wire installation at the very least) and connected to VCC via pull-up resistor to make the flashing software work:
![image](https://github.com/jdfr228/PS1-Disc-Based-Game-ID/assets/24398594/819dc2da-90aa-47e7-b609-c5a7b18e5016)

I also saw talk on an old PSXDEV thread where the flasher was brought up where people were connecting /CE to CPU pin 98 (/CS0) which looks like it connects to VSS through a resistor array, which I'm guessing would be fine if you don't have a pull-up resistor handy.  If you're able to try either please let me know how it goes!

Of course if you've flashed your own chip you can just directly write the modified BIOS to it.  I'm also working on fixes for games that have their SYSTEM.CNF file formatted differently (Ape Escape doesn't include the expected spaces for the Game ID string), and some hardware problems with the newest CPLD revision of MemCard Pro.  If you have an earlier FPGA one they seem to work fine but again let me know.

**Comment by p-b-p** [comment]
Hi @jdfr228,

I have tested the following scenarios, but unfortunately neither worked:

![20230614_110838](https://github.com/jdfr228/PS1-Disc-Based-Game-ID/assets/95297132/adfbc693-ef27-4af9-92e8-99f34838aef9)

**Case 1:**

- /WE is connected to /SWR0 (IC732 pin 120)
- /CE of the SST-39VF040 connected to VCC via 10kΩ resistor

When turnning on the console there is no image and the BIOS does not boot.

**Case 2:**

- /WE is connected to /SWR0 (IC732 pin 120)
- /CE of the SST-39VF040 connected to the motherboard

The console boots and loads the flasher CD and just before pressing L1 I've switched /CE to VCC via the resistor and screen goes black.

**Case 3**

- /WE connected to /SWR0 (IC732 pin 120)
- /CE to /CS0 (CPU pin 98)

The BIOS does not boot as well (same behaviour as in Case 1).

**Comment by jdfr228** [comment]
I'll ping megavolt again to see what he has to say on it.

I did see two different pinouts for IC732 though if that's what you're using for /WE:

SONY CXD2938Q pin 120 - /WE
SONY CXD2941R pin 120 - XHWR

Though looking at my PSOne schematics pdf with the CXD2938Q it's still labeled as XHWR and connected to SWR0 on the CPU, so maybe this is irrelevant and those are the same thing.

![image](https://github.com/jdfr228/PS1-Disc-Based-Game-ID/assets/24398594/6b0f17e4-0629-4b00-a0d3-c8777268b03b)


Also I'm happy to send you another 34VF040 chip or full board with pre-patched firmware on it once I fix the remaining bugs.  If you were able to flash the 39VF040 chip yourself I'm curious why you didn't go ahead and put the patched firmware on it, though I'm grateful for the flashing software testing we're getting out of it.

**Comment by p-b-p** [comment]
The problem was with the new XGecu T48 hardware for flashing SST39FV040 where the verification process fails because there is a bug in the software for this specific IC where the first byte of the image is replaced by the last one. Luckily this isn't causing any issues and the bios was starting correctly. That was the reason I wanted to test this flashing software and besides this way it is more convenient to reflash the bios without desoldering if necessary in the future.

**Comment by jdfr228** [comment]
To ping you to make sure you get a notification for this: @p-b-p 

I unfortunately decided to remove/deprecate the flasher program from the repo because you, myself, and other people who have tested it haven't gotten it to work so far.  If you want to keep trying with it feel free to message megavolt85, I've been able to reach him on the Dreamcast-talk forum.

I very recently released an updated version to fix several bugs which hopefully is final or near-final.  I still wanna offer to send you a reflashed chip or board if you'd like one, since I need more testers and feel like I led you astray with the software reflasher.

**Comment by VirtuaVespa** [comment]
> The problem was with the new XGecu T48 hardware for flashing SST39FV040 where the verification process fails because there is a bug in the software for this specific IC where the first byte of the image is replaced by the last one. Luckily this isn't causing any issues and the bios was starting correctly. That was the reason I wanted to test this flashing software and besides this way it is more convenient to reflash the bios without desoldering if necessary in the future.

Apologies if I'm not understanding this correctly but I've just flashed an SST39FV040 but **_not_** with this patch applied, just a regular bios file, with my T48 and it went smoothly, I had the 'Verify after' check box ticked and I didn't see any error messages. If I've not misunderstood could it be a software update has fixed this? The about says it's on ver: 12.63

**Comment by p-b-p** [comment]
> > The problem was with the new XGecu T48 hardware for flashing SST39FV040 where the verification process fails because there is a bug in the software for this specific IC where the first byte of the image is replaced by the last one. Luckily this isn't causing any issues and the bios was starting correctly. That was the reason I wanted to test this flashing software and besides this way it is more convenient to reflash the bios without desoldering if necessary in the future.
> 
> Apologies if I'm not understanding this correctly but I've just flashed an SST39FV040 but **_not_** with this patch applied, just a regular bios file, with my T48 and it went smoothly, I had the 'Verify after' check box ticked and I didn't see any error messages. If I've not misunderstood could it be a software update has fixed this? The about says it's on ver: 12.63

Yes, they fixed this issue with SW update and now you could flash SST-39VF040 without a problem.

**Comment by p-b-p** [comment]
> To ping you to make sure you get a notification for this: @p-b-p
> 
> I unfortunately decided to remove/deprecate the flasher program from the repo because you, myself, and other people who have tested it haven't gotten it to work so far. If you want to keep trying with it feel free to message megavolt85, I've been able to reach him on the Dreamcast-talk forum.
> 
> I very recently released an updated version to fix several bugs which hopefully is final or near-final. I still wanna offer to send you a reflashed chip or board if you'd like one, since I need more testers and feel like I led you astray with the software reflasher.

@jdfr228 Not needed to send me a pre-flashed chip. I plan to redesign my board with some kind of pcb socket so it is easy to swap after re-flash and then I could help you with the test of the different bios variants. Unfortunately lately I do not have enough free time to dedicate to this project.

**Comment by jdfr228** [comment]
> @jdfr228 Not needed to send me the pre-flashed chip. I plan to redesign my board with some kind of pcb socket so it is easy to swap after re-flash and then I could help you with the test of the different bios. Unfortunately lately I do not enough free time to dedicate to this project.

Sounds cool, and at least things are looking good for the project after the latest update, but yeah any testers are welcome, especially since I don't have any PAL consoles or anything to test with.

**Comment by VirtuaVespa** [comment]
> > > The problem was with the new XGecu T48 hardware for flashing SST39FV040 where the verification process fails because there is a bug in the software for this specific IC where the first byte of the image is replaced by the last one. Luckily this isn't causing any issues and the bios was starting correctly. That was the reason I wanted to test this flashing software and besides this way it is more convenient to reflash the bios without desoldering if necessary in the future.
> > 
> > 
> > Apologies if I'm not understanding this correctly but I've just flashed an SST39FV040 but **_not_** with this patch applied, just a regular bios file, with my T48 and it went smoothly, I had the 'Verify after' check box ticked and I didn't see any error messages. If I've not misunderstood could it be a software update has fixed this? The about says it's on ver: 12.63
> 
> Yes, they fixed this issue with SW update and now you could flash SST-39VF040 without a problem.

Thank you for confirming this!
